src/main/java/org/opensearch/ad/transport/InsightsJobTransportAction.java

```diff
@@ -0,0 +1,156 @@
+        Settings settings,
+        NamedXContentRegistry xContentRegistry,
+        ADIndexManagement indexManagement
+    ) {
+        super(InsightsJobAction.NAME, transportService, actionFilters, InsightsJobRequest::new);
+        this.client = client;
+        this.jobHandler = new InsightsJobActionHandler(
+            client,
+            xContentRegistry,
+            indexManagement,
+            AnomalyDetectorSettings.AD_REQUEST_TIMEOUT.get(settings)
+        );
+    }
+
+    @Override
+    protected void doExecute(Task task, InsightsJobRequest request, ActionListener<InsightsJobResponse> listener) {
+        if (request.isStartOperation()) {
+            handleStartOperation(request, listener);
+        } else if (request.isStatusOperation()) {
+            handleStatusOperation(request, listener);
+        } else if (request.isStopOperation()) {
+            handleStopOperation(request, listener);
+        } else if (request.isResultsOperation()) {
+            handleResultsOperation(request, listener);
+        } else {
+            listener.onFailure(new IllegalArgumentException("Unknown operation"));
+        }
+    }
+
+    private void handleStartOperation(InsightsJobRequest request, ActionListener<InsightsJobResponse> listener) {
+        log.info("Starting insights job with frequency: {}", request.getFrequency());
+
+        jobHandler.startInsightsJob(request.getFrequency(), listener);
+    }
+
+    private void handleStatusOperation(InsightsJobRequest request, ActionListener<InsightsJobResponse> listener) {
+        jobHandler.getInsightsJobStatus(listener);
+    }
+
+    private void handleStopOperation(InsightsJobRequest request, ActionListener<InsightsJobResponse> listener) {
+        jobHandler.stopInsightsJob(listener);
+    }
+
+    private void handleResultsOperation(InsightsJobRequest request, ActionListener<InsightsJobResponse> listener) {
+        try {
+            BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
+
+            if (request.getDetectorId() != null && !request.getDetectorId().isEmpty()) {
+                boolQuery.must(QueryBuilders.termQuery("doc_detector_ids", request.getDetectorId()));
+            }
+
+            if (request.getIndex() != null && !request.getIndex().isEmpty()) {
+                boolQuery.must(QueryBuilders.termQuery("doc_indices", request.getIndex()));
+            }
+
+            if (!boolQuery.hasClauses()) {
+                boolQuery.must(QueryBuilders.matchAllQuery());
+            }
+
+            SearchRequest searchRequest = new SearchRequest(ADCommonName.INSIGHTS_RESULT_INDEX_ALIAS)
+                .source(
+                    new SearchSourceBuilder()
+                        .query(boolQuery)
+                        .from(request.getFrom())
+                        .size(request.getSize())
+                        .sort("generated_at", SortOrder.DESC)
+                );
+
+            client.search(searchRequest, ActionListener.wrap(searchResponse -> {
```

Comment:
Do you need to add backend role filtering before search? Please add security tests with backend role filtering on.

Reply:
We need to do tenant-isolated search, but not necessarily backend role filtering here. For insights generation, it's a background job, so followed existing pattern to use InjectSecurity directly for background work, just impersonate the stored user via InjectSecurity, then execute search directly. For user-facing search APIs like search anomaly result transport action, AD reads the current user from thread context and then adds backend role filtering. 

Adding security tests in the next revision

Reply:
So we have three kinds of auth now:
1. fgac role
2. backend-role filtering
3. resource sharing.

We will need to cover all three of them.